process_report/invoices/pi_specific_invoice.py

```diff
@@ -92,20 +92,44 @@ def _get_pi_dataframe(self, data, pi):
             if column_name in pi_projects.columns:
                 column_sums.append(pi_projects[column_name].sum())
                 sum_columns_list.append(column_name)
-        pi_projects.loc[len(pi_projects)] = (
-            None  # Adds a new row to end of dataframe initialized with None
+
+        # Create a new row with proper dtypes
+        new_row = {col: None for col in pi_projects.columns}
+
+        # Add Invoice Month column if it doesn't exist
+        if invoice.INVOICE_DATE_FIELD not in pi_projects.columns:
+            pi_projects[invoice.INVOICE_DATE_FIELD] = None
+            new_row[invoice.INVOICE_DATE_FIELD] = None
+
+        new_row[invoice.INVOICE_DATE_FIELD] = "Total"
+        for col, val in zip(sum_columns_list, column_sums):
+            new_row[col] = val
+
+        # Convert all columns to object type before concatenation to avoid dtype warnings
+        pi_projects = pi_projects.astype("object")
+
+        # Add the totals row
+        pi_projects = pandas.concat(
+            [pi_projects, pandas.DataFrame([new_row]).astype("object")],
+            ignore_index=True,
         )
-        pi_projects.loc[pi_projects.index[-1], invoice.INVOICE_DATE_FIELD] = "Total"
-        pi_projects.loc[pi_projects.index[-1], sum_columns_list] = column_sums
 
         # Add dollar sign to certain columns
         for column_name in self.DOLLAR_COLUMN_LIST:
             if column_name in pi_projects.columns:
                 pi_projects[column_name] = pi_projects[column_name].apply(
-                    lambda data: data if pandas.isna(data) else f"${data}"
+                    lambda data: data if pandas.isna(data) else f"${float(data)}"
                 )
 
-        pi_projects.fillna("", inplace=True)
+        # Convert all numeric columns to strings before filling NA values
+        # This prevents dtype incompatibility warnings
+        for col in pi_projects.columns:
+            # First ensure all columns are object type
+            if pi_projects[col].dtype.name.startswith(("float", "int")):
+                pi_projects[col] = pi_projects[col].astype("object")
```

Comment:
You should cast to Pandas' built-in [`StringDType`](https://pandas.pydata.org/docs/reference/api/pandas.StringDtype.html) instead. It is stricter than `object` and has other compatibilities with Pandas. 

It should also be fine to cast the entire dataframe to `StringDtype`. The entire dataframe will be rendered as html text anyways:

```suggestion python
pi_projects = pi_projects.astype(invoice.STRING_FIELD_TYPE)
pi_projects.fillna("", inplace=True)
```